### PR TITLE
Implement linting and fix linter errors

### PR DESCRIPTION
**Pull request by SirCmpwn**
Notes:

* I have left unfixed all of the linting errors related to declaring the types of our props. I'll be fixing that in a seperate pull request.
* Because of that, I have not yet made the changes that will fail a Travis build if the linter fails.

All unit tests pass and I tested everything in the UI manually to make sure I didn't break anything.

Ready for review+merge.

### CODE REVIEW

File: src/layouts/OAuth.js [hunk 1]
```diff
@@ -11,16 +11,16 @@ class OAuthCallbackPage extends Component {
     const returnTo = location.query['return'];
 
     if (code) {
-      let xhr = new XMLHttpRequest();
+      const xhr = new XMLHttpRequest();
```
Comment (eatonphil) [inline src/layouts/OAuth.js]: The linter catches when something can be a const?

Comment (SirCmpwn) [inline src/layouts/OAuth.js]: Yep.

Comment (SirCmpwn) [inline src/layouts/OAuth.js]: It catches a fuckton of stuff.

**Comment by eatonphil** [comment]
This is awesome. Good stuff. Ready to merge?

**Comment by SirCmpwn** [comment]
Yes.

**Comment by na3d** [comment]
👍

**Comment by eatonphil** [comment]
:+1:

**Pull request merged**